src/udev/udevadm-info.c

```diff
@@ -982,55 +1135,64 @@ int info_main(int argc, char *argv[], void *userdata) {
                         if (!path)
                                 return log_oom();
 
-                        r = strv_consume(&devices, path);
+                        r = strv_consume(&arg_devices, path);
                         if (r < 0)
                                 return log_oom();
                         break;
                 }
 
                 case 'q':
-                        action = ACTION_QUERY;
+                        arg_action = ACTION_QUERY;
                         if (streq(optarg, "property") || streq(optarg, "env"))
-                                query = QUERY_PROPERTY;
+                                arg_query = QUERY_PROPERTY;
                         else if (streq(optarg, "name"))
-                                query = QUERY_NAME;
+                                arg_query = QUERY_NAME;
                         else if (streq(optarg, "symlink"))
-                                query = QUERY_SYMLINK;
+                                arg_query = QUERY_SYMLINK;
                         else if (streq(optarg, "path"))
-                                query = QUERY_PATH;
+                                arg_query = QUERY_PATH;
                         else if (streq(optarg, "all"))
-                                query = QUERY_ALL;
+                                arg_query = QUERY_ALL;
```

Comment:
Maybe better to use a string table?

Reply:
Makes sense.

Reply:
Or does it? The code would still have to handle `streq(optarg, "env")` separately, because "property" and "env" are synonyms and that cannot be expressed in the table's initializer. As a result, the code probably wouldn't be any simpler than it is now. OTOH, maybe it's better to handle the deprecated query type (`env`) separately from the rest? I don't know...

Reply:
Not a strong opinion, feel free to ignore the suggestion.

Reply:
I tried it and I like the result, so it's in.